pandas/core/series.py

```diff
@@ -2559,7 +2568,11 @@ def _reindex_indexer(self, new_index, indexer, copy):
             return self
 
         # be subclass-friendly
-        new_values = algorithms.take_1d(self.get_values(), indexer)
+        if isinstance(self.values, ExtensionArray):
```

Comment:
`self._values`? 
Or did we decide that `.values` will return the extension array (also for our own future extension types Period, DatetimeTZ, ..)

Reply:
We keep putting off that discussion to a later date :) Will use `._values` for now.

Reply:
rather than this. pls just do this in algorithms.take where this is currently handled (IOW there is dispatch there to categorical, datetime, sparse etc). Sure that should be unwound, but adding addition if/then here is just more technical debt.

Reply:
So, I *really* want to avoid defining a `get_values` method on `EA`. We've been turning up all sorts of performance issues with `Categorical.get_values` being called. On master we currently materialize the categories with doing `Series[categorical].align(Series[categorical])`.

I will see how hard it is to make `algos.take_nd` handle sparse correctly, so that we can just pass `self._values` here...